### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by G-jonathan**
Work of the week.
The main method has been reduced to a maximum by creating two classes: 
- One which reads the file and returns an array, 
- One which browse the array, count the occurrences of each symptoms and writes the result to a file.

I commented the code with personal comments and javadoc.
Addition of the "result" file on the remote repository.

### CODE REVIEW

File: Project02Eclipse/src/com/hemebiotech/analytics/ReadSymptomDataFromFile.java [hunk 1]
```diff
@@ -31,7 +36,7 @@ public ReadSymptomDataFromFile (String filepath) {
 				BufferedReader reader = new BufferedReader (new FileReader(filepath));
 				String line = reader.readLine();
 				
-				while (line != null) {
+				while (line != null) { // we browse the file line by line then we add the content of each line in the result array
```
Comment (Asavarkhul) [inline Project02Eclipse/src/com/hemebiotech/analytics/ReadSymptomDataFromFile.java]: The comments are only accepted on functions/properties definitions. If you need to comment a code line, it means that it's not correctly coded. You can remove this one.

File: Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		this.arraySymptoms = arraySymptoms;
+	}
+	
+	/**
+	 * Extract the differents symptoms and count their occurrences
+	 * Generates a file containing the list of symptoms and the number of times they appear in the file
+	 * @throws IOException 
+	 */
+	public void WriteSymptoms() throws IOException {
+		FileWriter writer = new FileWriter("result.out");
```
Comment (Asavarkhul) [inline Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java]: Avoid declaration inside a function. You should have this one as a property on class level, and inject it in the constructor. (like the `arraySymptoms` property you did).

File: Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		ArrayList<String> deduplication = new ArrayList<String>();
```
Comment (Asavarkhul) [inline Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java]: From there, all you code can be simplified in 3 lines 😇
Tips -> look at the `Collection` java documentation. There is a `frequency ` method, which could help you a lot.

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#frequency(java.util.Collection,%20java.lang.Object)

**Review by Asavarkhul — CHANGES_REQUESTED** [review]

**Comment by Asavarkhul** [comment]
Good job anyway 👍

File: Project02Eclipse/src/com/hemebiotech/analytics/ReadSymptomDataFromFile.java (hunk 1 above)
Comment (G-jonathan) [reply Project02Eclipse/src/com/hemebiotech/analytics/ReadSymptomDataFromFile.java]: Ok, i imagine it's also valid for the other files.
It's done.

File: Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java (hunk 2 above)
Comment (G-jonathan) [reply Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java]: It's done

File: Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java (hunk 3 above)
Comment (G-jonathan) [reply Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java]: It's done but there are a little more than 3 lines.
you will tell me tomorow  if it's what you expected :)

**Pull request merged**